src/material/select/select.ts

```diff
@@ -553,10 +556,18 @@ export class MatSelect
       );
     }
 
-    return this._ngZone.onStable.pipe(
-      take(1),
-      switchMap(() => this.optionSelectionChanges),
-    );
+    return new Observable(subscriber => {
```

Comment:
What the previous code was working around was that somebody can try to subscribe before `this.options` is defined in `ngAfterContentInit`. I think that we can simplify this even more by doing something like this instead:

```
private initialize = new Subject();

readonly optionSelectionChanges = defer(() => {
  if (options) {
      ....
   }

   return this.initialized.switchMap(...);
});

ngAfterContentInit() {
  this.initialized.next();
  this.initialized.complete();
}
```